### ISSUE TITLE
Adapter should extend from AbstractAdapter

### ISSUE THREAD

**Issue by panvid** [laminas/laminas-cache-storage-adapter-blackhole#6]
Due https://github.com/laminas/laminas-cache/pull/21 all Adapters, which should be created via *StorageFactory* should extend the *AbstractAdapter* to not trigger a deprecation message.

This happens since Release *2.10.0* of *laminas/laminas-cache*, see: https://github.com/laminas/laminas-cache/releases/tag/2.10.0

**Comment by boesing** [comment]
@dpauli Thanks for reporting this. Could you provide some code to reproduce your issue?

Because having all adapters extend `AbstractAdapter` is not correct.
Since `laminas-cache` v2.10, all adapters created by the `AdapterPluginManager` *should* implement the new `PluginAwareInterface` *if* a configuration is passed to the `StorageFactory` which contains a `plugins`-Key.

So, since the `BlackHole` adapter never implemented the `EventsCapableInterface`, there was a `RuntimeException` in v2.9 already, when `StorageFactory` was used for the `BlackHole` adapter with provided `plugins`-Key.

**To summarize:** the blackhole adapter is not capable of plugins and thus wont trigger any deprecation.

So having a config you are passing to the `StorageFactory` which triggers the deprecation message would be great.
Thanks in advance.

**Comment by panvid** [comment]
Thanks for the fast feedback. I checked my assumption with *2.10.0* `laminas-cache`-Release and can say: Their is no problem with this library.

This issue can be closed :)
